lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -410,31 +410,72 @@ bool SinkCode::run(Function *F, const CompilationContext &cctx) {
   // For each node:
   for (auto &N : nodes) {
     auto *node = &N;
-    // Sink Transpose below batch normalization nodes:
+    // Sink Reshape/Transpose below BatchNormalization.
     if (auto *BN = dyn_cast<BatchNormalizationNode>(node)) {
-      auto *TR = dyn_cast<TransposeNode>(BN->getInput());
 
-      if (!TR) {
+      // Sink Reshape below BatchNormalization.
+      if (auto *RS = dyn_cast<ReshapeNode>(BN->getInput())) {
+        auto inDims = RS->getInput().dims();
+        auto outDims = RS->getResult().dims();
+
+        // Reshape should not change the BatchNorm ChannelIdx dimensions.
+
+        // Check for NH[W]C layout.
+        if (outDims[BN->getChannelIdx()] == outDims.size() - 1) {
+          if (inDims.size() - 1 != outDims.size() - 1) {
+            continue;
+          }
+        }
```

Comment:
I'm a little confused -- shouldn't this be something like 
```suggestion
        if (BN->getChannelIdx() == outDims.size() - 1 && 
            inDims[size() - 1] == outDims[size() -1]) {
          continue;
        }
```

Reply:
@jfix71, Thanks for pointing put the mistake. I have fixed this for NH[W]C layout.